src/Raven.Client/Documents/Identity/AsyncHiLoIdGenerator.cs

```diff
@@ -76,7 +77,18 @@ public RangeValue(long min, long max)
         /// <returns></returns>
         public Task<string> GenerateDocumentIdAsync(object entity)
         {
-            return NextIdAsync().ContinueWith(task => GetDocumentIdFromId(task.Result));
+            return NextIdAsync().ContinueWith(task =>
+            {
+                try
+                {
+                    return GetDocumentIdFromId(task.Result);
+                }
+                catch (Exception e)
```

Comment:
This obfuscate the actual exception stack trace. 
I don't follow why this is needed?

Reply:
done